### PR TITLE
ENTERPRISE-724: Allie / updating ip restriction docs

### PR DESCRIPTION
**Pull request by nm0s**
## What are you changing in this pull request and why?

Updating docs to reflect that service tokens will now be blocked by ip restrictions. And BC customers with Gitlab or ADO integrations must add their respective integration ips to their ip rules

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/cloud/git/connect-gitlab.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ The application form in GitLab should look as follows when completed:
 
 Click **Save application** in GitLab, and GitLab will then generate an **Application ID** and **Secret**. These values will be available even if you close the app screen, so this is not the only chance you have to save them.
 
+If you're a Business Critical customer using IP restrictions ensure you've added the appropriate Gitlab CIDRs to your IP restriction rules, or else the Gitlab connection will fail.
```
Comment (nm0s) [inline website/docs/docs/cloud/git/connect-gitlab.md]: @saimaddali I've added these lines to the bottom of the gitlab and ADO integration sections - would you rather these be in a more visible place?

**Review by BrJan — APPROVED** [review]

File: website/docs/docs/cloud/git/connect-azure-devops.md [hunk 2]
```diff
@@ -23,3 +23,4 @@ To connect Azure DevOps in dbt Cloud:
 2. dbt Cloud developers need to [personally authenticate with Azure DevOps](/docs/cloud/git/authenticate-azure) from dbt Cloud.
 
 
+If you're a Business Critical customer using IP restrictions ensure you've added the appropriate Azure DevOps CIDRs to your IP restriction rules, or else the Azure DevOps connection will fail.
```
Comment (ajaythomas) [inline website/docs/docs/cloud/git/connect-azure-devops.md]: > using IP restrictions ensure

nit: a `,` before ensure.
Can you also link IP restrictions back to `website/docs/docs/cloud/secure/ip-restrictions.md` docs

Comment (ajaythomas) [reply website/docs/docs/cloud/git/connect-azure-devops.md]: Similar below.

File: website/docs/docs/cloud/git/connect-gitlab.md (hunk 1 above)
Comment (ajaythomas) [reply website/docs/docs/cloud/git/connect-gitlab.md]: Let's also add it to the ip restriction docs (`website/docs/docs/cloud/secure/ip-restrictions.md`)

**Review by ajaythomas — APPROVED** [review]

File: website/docs/docs/cloud/secure/ip-restrictions.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ To configure IP restrictions, go to **Account Settings** → **IP Restrictions**
 - Deny IPs flagged by the Security team
 - Allow only VPN traffic but make an exception for contractors’ IP addresses
 
-IP restrictions will block all user requests done via the API (via personal user token) and the UI. Service tokens are exempt from IP restrictions and can still make requests to dbt Cloud API.
+IP restrictions will block all service tokens, user requests done via the API (via personal user token), and the UI.
```
Comment (ajaythomas) [inline website/docs/docs/cloud/secure/ip-restrictions.md]: IP restrictions will block all service tokens requests, user requests done via the API (via personal user token), and the UI, if they are coming from blocked IP addresses.

Comment (ajaythomas) [reply website/docs/docs/cloud/secure/ip-restrictions.md]: This is for you can add a note about integrations using serviceTokens. 
"For any versionControl system integrations inbound into dbt Cloud, ensure their IP addresses are added to the allowed list. Examples: Gitlab, ADO and Github"

File: website/docs/docs/cloud/secure/ip-restrictions.md [hunk 4]
```diff
@@ -19,7 +19,9 @@ To configure IP restrictions, go to **Account Settings** → **IP Restrictions**
 - Deny IPs flagged by the Security team
 - Allow only VPN traffic but make an exception for contractors’ IP addresses
 
-IP restrictions will block all user requests done via the API (via personal user token) and the UI. Service tokens are exempt from IP restrictions and can still make requests to dbt Cloud API.
+IP restrictions will block all service tokens, user requests done via the API (via personal user token), and the UI, if they are coming from blocked IP addresses.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/secure/ip-restrictions.md]:
```suggestion
IP restrictions will block all service tokens, user requests done via the API (via personal user token), and the UI if they come from blocked IP addresses.
```

File: website/docs/docs/cloud/secure/ip-restrictions.md [hunk 5]
```diff
@@ -19,7 +19,9 @@ To configure IP restrictions, go to **Account Settings** → **IP Restrictions**
...
+
+For any version control system integrations inbound into dbt Cloud, ensure their IP addresses are added to the allowed list. Examples: Gitlab, ADO and Github.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/secure/ip-restrictions.md]:
```suggestion
For any version control system integrations (Github, Gitlab, ADO, etc.) inbound into dbt Cloud, ensure their IP addresses are added to the allowed list.
```

File: website/docs/docs/cloud/git/connect-gitlab.md [hunk 6]
```diff
@@ -71,6 +71,8 @@ The application form in GitLab should look as follows when completed:
 
 Click **Save application** in GitLab, and GitLab will then generate an **Application ID** and **Secret**. These values will be available even if you close the app screen, so this is not the only chance you have to save them.
 
+If you're a Business Critical customer using [IP restrictions](/docs/cloud/secure/ip-restrictions) ensure you've added the appropriate Gitlab CIDRs to your IP restriction rules, or else the Gitlab connection will fail.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/git/connect-gitlab.md]:
```suggestion
If you're a Business Critical customer using [IP restrictions](/docs/cloud/secure/ip-restrictions), ensure you've added the appropriate Gitlab CIDRs to your IP restriction rules, or else the Gitlab connection will fail.
```

**Review by matthewshaver — APPROVED** [review]

**Pull request merged**